repos/system_upgrade/el7toel8/actors/ziplcheckbootentries/libraries/ziplcheckbootentries.py

```diff
@@ -0,0 +1,134 @@
+            reporting.Flags([reporting.Flags.INHIBITOR])
+        ])
+
+
+def extract_kernel_version(kernel_img_path):
+    """
+    Extracts the kernel version out of the given image path.
+
+    The extraction logic is designed to closely mimick the logic Zipl configuration to BLS
+    conversion script works, so that it is possible to identify the possible issues with kernel
+    images.
+
+    :param str kernel_img_path: The path to the kernel image.
+    :returns: Extracted kernel version from the given path
+    :rtype: str
+    """
+
+    if 'vmlinuz-' in kernel_img_path:
+        fragments = kernel_img_path.rsplit('/vmlinuz-', 1)
+        # Mimick bash substitution used in the conversion script
+        return fragments[1] if len(fragments) > 1 else fragments[0]
+
+    fragments = kernel_img_path.rsplit('/', 1)
+    return fragments[1] if len(fragments) > 1 else fragments[0]
+
+
+def inhibit_if_entries_share_kernel_version(bootloader_config):
+    """
+    Inhibits the upgrade if there are boot entries sharing the same kernel image version.
+
+    The logic of identification whether the images are the same mimics the zipl-switch-to-blscfg, as it fails
+    to perform the conversion if there are entries with the same kernel image.
+
+    :param SourceBootloaderConfiguration bootloader_config: The configuration of the source boot loader.
+    """
+
+    used_kernel_versions = defaultdict(list)  # Maps images to the boot entries in which they are used
+    for boot_entry in bootloader_config.entries:
+        if is_rescue_entry(boot_entry):
+            # Rescue entries are handled differently and their images should not cause naming collisions
+            continue
+
+        kernel_version = extract_kernel_version(boot_entry.kernel_image)
+        used_kernel_versions[kernel_version].append(boot_entry)
+
+    versions_used_multiple_times = []
+    for version, version_boot_entries in used_kernel_versions.items():
+        if len(version_boot_entries) > 1:
+            # Keep the information about entries for the report
+            versions_used_multiple_times.append((version, version_boot_entries))
+
+    if versions_used_multiple_times:
+        problematic_entries_details = ''
+        for version, version_boot_entries in versions_used_multiple_times:
+            entry_titles = ['"{0}"'.format(entry.title) for entry in version_boot_entries]
+            problematic_entries_details += '{0}{1} (found in entries: {2})'.format(
+                    FMT_LIST_SEPARATOR,
+                    version,
+                    ', '.join(entry_titles)
+            )
+
+        reporting.create_report([
+            reporting.Title('Boot entries sharing the same kernel version found.'),
+            reporting.Summary(
+                'The zipl configuration file {0} contains boot entries sharing the same kernel version: {1}'
+                .format(ZIPL_CONFIG_PATH, problematic_entries_details)
```

Comment:
@MichalHe add please more info into the summary explaining why this is the problem. e.g. mention something like: It's not possible to migrate the configuration to the BLS format correctly in such a case, which causes issues with the bootloader after the upgrade.